src/main/java/org/hyperledger/bela/converter/RocksDBKeyValueStorageConverterFactory.java

```diff
@@ -64,26 +66,24 @@ public KeyValueStorage create(
             segment, create(segments, commonConfiguration, metricsSystem));
     }
 
-    /**
-     * This method should be refactored, we ignore the passed in list of segments in favor of
-     * the list of segments passed into this Factory constructor.
-     *
-     * @param requestedSegments IGNORED list of segments identifiers that comprise the created segmented storage.
-     * @param commonConfiguration common configuration available to plugins, in a populated state.
-     * @param metricsSystem metrics component for recording key-value storage events.
-     */
     @Override
     public SegmentedKeyValueStorage create(List<SegmentIdentifier> requestedSegments,
         BesuConfiguration commonConfiguration, MetricsSystem metricsSystem) throws StorageException {
         if (requiresInit()) {
             init(commonConfiguration);
         }
+
+        // create segmented storage for the distinct set of requested and detected segments
+        var allSegments = Streams.concat(segments.stream(), requestedSegments.stream())
+            .distinct()
+            .collect(Collectors.toList());
```

Comment:
nit: can use toList to make this simpler

```suggestion
            .toList();
```